plugins/datastreamer_plugin/fastdds/ReaderHandler.cpp

```diff
@@ -93,11 +87,9 @@ void ReaderHandler::on_data_available(
     ReturnCode_t read_ret = RETCODE_OK;
 
     // Non-fixed size types require data to be recreated (e.g. to avoid having sequence members from previous samples)
-    if (!static_type_)
-    {
-        DynamicDataFactory::get_instance()->delete_data(data_);
-        data_ = DynamicDataFactory::get_instance()->create_data(type_);
-    }
+    DynamicDataFactory::get_instance()->delete_data(data_);
+    data_ = DynamicDataFactory::get_instance()->create_data(type_);
```

Comment:
I think so. For non static types like unions or maps, active members are determined in each data sample. I think we need to create a new dynamic data in these cases to avoid corrupting new data with previous data samples.